kernel/src/net/socket/ip/stream/mod.rs

```diff
@@ -627,40 +601,14 @@ impl Socket for StreamSocket {
 
     fn set_option(&self, option: &dyn SocketOption) -> Result<()> {
         let mut options = self.options.write();
+        match options.socket.set_option(option) {
```

Comment:
Add an empty line before this line. Locking options should be a separate block of code, since the locked options will be used multiple times below.